File: Excel/666 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/666 Fill in Blanks.xlsx"
input = read_excel(path, range = "A2:D11")
test  = read_excel(path, range = "F2:I11")

fill_missing_values <- function(row) {
  na_index <- which(is.na(row[-1])) + 1
  if (length(na_index) > 0) {
    row[na_index] <- (row$Total - sum(row[-1], na.rm = TRUE)) / length(na_index)
  }
  return(row)
}

result = input
for (i in 1:nrow(input)) {
  result[i,] <- fill_missing_values(input[i,])
}

all.equal(result, test)
#> [1] TRUE

File: Excel/600-699/666/666 Challenge.py
import pandas as pd

path = "666 Fill in Blanks.xlsx"
input = pd.read_excel(path, usecols="A:D", skiprows=1, nrows=9)
test = pd.read_excel(path, usecols="F:I", skiprows=1, nrows=9).rename(columns=lambda x: x.split('.')[0])

def fill(input_df, rn):
    row = input_df.iloc[rn]
    na_positions = row[row.isna()].index
    if not na_positions.empty:
        row[na_positions] = (row[0] - row[1:].sum()) / len(na_positions)
    return row


result = input.apply(lambda row: fill(input, row.name).astype('int64'), axis=1)
print(result.equals(test))  # True